File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
	"strings"
)

// Convert creates raindrop sounds based on the input number.
// If the number is evenly divisible by 3, it adds "Pling"
// If the number is evenly divisible by 5, it adds "Plang"
// If the number is evenly divisible by 7, it adds "Plong"
// Otherwise, it returns the number as string
func Convert(n int) string {
	var sb strings.Builder

	if isDivisbleBy(n, 3) {
		sb.WriteString("Pling")
	}

	if isDivisbleBy(n, 5) {
		sb.WriteString("Plang")
	}

	if isDivisbleBy(n, 7) {
		sb.WriteString("Plong")
	}

	if result := sb.String(); result != "" {
		return result
	}

	return fmt.Sprint(n)
}

func isDivisbleBy(a, b int) bool {
	return a%b == 0
}

File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(number_of_drops)
    result = ""

    result += "Pling" if number_of_drops % 3 == 0
    result += "Plang" if number_of_drops % 5 == 0
    result += "Plong" if number_of_drops % 7 == 0

    result == "" ? number_of_drops.to_s : result
  end
end
